src/vellum/workflows/inputs/dataset_row.py

```diff
@@ -45,3 +45,57 @@ def serialize_inputs(self, inputs: Union[BaseInputs, Dict[str, Any]]) -> Dict[st
                 result[input_descriptor.name] = value
 
         return result
+
+    @field_serializer("node_output_mocks")
+    def serialize_node_output_mocks(
+        self, node_output_mocks: Optional[Sequence[Union[BaseOutputs, MockNodeExecution]]]
+    ) -> Optional[List[Dict[str, Any]]]:
+        """
+        Custom serializer for node_output_mocks that normalizes both BaseOutputs and MockNodeExecution
+        to a consistent dict format with node_id, when_condition, and then_outputs.
+
+        Args:
+            node_output_mocks: Optional sequence of BaseOutputs or MockNodeExecution instances
+
+        Returns:
+            List of normalized mock execution dicts, or None if input is None
+        """
+        if node_output_mocks is None:
+            return None
+
+        result = []
+        for mock in node_output_mocks:
+            if isinstance(mock, MockNodeExecution):
+                node_id = mock.then_outputs.__class__.__parent_class__.__id__
+                then_outputs_dict = {
+                    descriptor.name: value
+                    for descriptor, value in mock.then_outputs
+                    if descriptor.name and not descriptor.name.startswith("__")
+                }
+                normalized_mock = {
+                    "node_id": node_id,
+                    "when_condition": {
+                        "type": "CONSTANT_VALUE",
+                        "value": {"type": "JSON", "value": True},
+                    },
+                    "then_outputs": then_outputs_dict,
+                }
+            else:
```

Comment:
In this case, we should construct a `MockNodeExecution` from a `BaseOutputs`, then call `.model_dump()` like above